packages/react-components/src/components/chart/events/useHandleChartEvents.ts

```diff
@@ -75,61 +78,68 @@ export const useHandleChartEvents = (chartRef: MutableRefObject<EChartsType | nu
     // Define event handler function so that we can remove this specific behavior on re-render
 
     // When mouse is down only update when shift is also down
-    const mouseDownEventHandler = () => {
-      setMouseDown(true);
+    // if the brush zoom is selected, prevent event propagation
+    const mouseDownEventHandler = (e: any) => {
       if (shiftDown) {
-        setOnDragStart(chart);
+        setOnPanStart(chart);
+      }
+      if (isBrushZooming) {
+        e.stop();
       }
     };
 
     chart?.getZr().on('mousedown', mouseDownEventHandler);
 
     // When mouse is released only update when shift is also down
     const mouseUpEventHandler = () => {
-      setMouseDown(false);
       if (shiftDown) {
-        setOnDragEnd(chart);
+        setOnPanEnd(chart);
       }
     };
 
     chart?.getZr().on('mouseup', mouseUpEventHandler);
 
     // Most important to update dragging events on mouse move
     const mouseMoveEventHandler = () => {
-      const isGrabbing = mouseDown && shiftDown;
-
       // Only update tooltip when state changed
       let stateChanged = false;
-      if (isGrabbing !== prevIsGrabbing) {
-        setPrevIsGrabbing(isGrabbing);
+      if (isPanning !== prevIsPanning) {
+        setPrevIsPanning(isPanning);
         stateChanged = true;
       } else {
         stateChanged = false;
       }
 
       // Overwrite cursor style on every move of the mouse
       // Update tooltip only once, on grab state change
-      if (isGrabbing) {
+      if (isPanning) {
+        setIsPanning(true);
         chart?.getZr().setCursorStyle('grabbing');
         if (stateChanged) {
-          setOnDragStart(chart);
+          setOnPanStart(chart);
         }
-      } else if (shiftDown) {
+      } else if (shiftDown && !isPanning) {
         chart?.getZr().setCursorStyle('grab');
         if (stateChanged) {
-          setOnDragEnd(chart);
+          setOnPanEnd(chart);
         }
       }
     };
 
     chart?.getZr().on('mousemove', mouseMoveEventHandler);
 
+    const zoomClickHandler = (e: any) => {
```

Comment:
you cant use any :)

Reply:
yeah i know 😢 looking into what event this would be bc echarts defines it as:
`WithThisType<EvtDef[EvtNm], CbThis<Ctx, this>>`